frontend/src/component/filter/Filters.tsx

```diff
@@ -0,0 +1,147 @@
+import { useEffect, useState, VFC } from 'react';
+import { Box, styled } from '@mui/material';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import AddFilterButton from './AddFilterButton';
+import { FilterDateItem } from 'component/common/FilterDateItem/FilterDateItem';
+import { FilterItem, FilterItemParams } from './FilterItem/FilterItem';
+
+const StyledBox = styled(Box)(({ theme }) => ({
+    display: 'flex',
+    padding: theme.spacing(2, 3),
+    gap: theme.spacing(1),
+    flexWrap: 'wrap',
+}));
+
+export type FilterItemParamHolder = Record<
+    string,
+    FilterItemParams | null | undefined
+>;
+
+interface IFilterProps {
+    state: FilterItemParamHolder;
+    onChange: (value: FilterItemParamHolder) => void;
+    availableFilters: IFilterItem[];
+}
+
+type IBaseFilterItem = {
+    label: string;
+    options: {
+        label: string;
+        value: string;
+    }[];
+    filterKey: string;
+};
+
+type ITextFilterItem = IBaseFilterItem & {
+    singularOperators: [string, ...string[]];
+    pluralOperators: [string, ...string[]];
+};
+
+type IDateFilterItem = IBaseFilterItem & {
+    dateOperators: [string, ...string[]];
+};
+
+export type IFilterItem = ITextFilterItem | IDateFilterItem;
+
+export const Filters: VFC<IFilterProps> = ({
```

Comment:
This is single reusable component that can be used with every table.

Reply:
if it's gonna be used a lot we need a test. Is that gonna be a followup PR?

Reply:
Yes